src/utils/workflow-utils.ts

```diff
@@ -216,6 +218,20 @@ export const generateTaskSpec = (
                         delete draftSpec.bindings[existingBindingIndex].disable;
                     }
 
+                    if (
+                        entityType === 'materialization' &&
+                        incompatibleSchemaChanges &&
+                        hasLength(incompatibleSchemaChanges[bindingUUID])
+                    ) {
+                        draftSpec.bindings[
+                            existingBindingIndex
+                        ].onIncompatibleSchemaChange =
+                            incompatibleSchemaChanges[bindingUUID];
+                    } else if (entityType === 'materialization') {
+                        delete draftSpec.bindings[existingBindingIndex]
+                            .onIncompatibleSchemaChange;
+                    }
```

Comment:
What is driving storing this outside of `meta` ?

Reply:
Honestly? I was on the fence on whether this piece of state should be part of the binding metadata or be stored in a separate slice. Presently, all materialization-specific binding properties are stored in slices and, for ease of implementation, it made sense for `onIncompatibleSchemaChange` to follow suit. Will remove the new slice entirely and fold the state into the binding store since that _sounds_ like the direction you are leaning toward.